### PR TITLE
[StableHloExt] Add ReifyRankedShapedTypeOpInterface for ReduceWindowOp

### PR DESCRIPTION
**Pull request by yizhuoz004**
This change adds a ReifyRankedShapedTypeOpInterface interface for ReduceWindowOp to enable dynamic shape inference.

### CODE REVIEW

File: mlir-tensorrt/test/Dialect/StableHloExt/reify-ranked-shaped-type.mlir [hunk 1]
```diff
@@ -96,3 +96,86 @@ func.func @dynamic_input(%arg0: tensor<?x?x?x?xf32>, %arg1: tensor<1x1x1024x1024
 //   CHECK-DAG: %[[v6:.+]] = arith.addi %[[v5]], %[[c1]] : index
 //   CHECK-DAG: %[[v7:.+]] = arith.maxsi %[[v6]], %[[c0]] : index
 //   CHECK-DAG: return %[[dim]], %[[c256]], %[[v3]], %[[v7]] :
+
+// -----
+
+func.func @refine_reduce_window(%arg0 : tensor<4x3x1024x1024xf32>)
+        -> (index, index, index, index) {
+  %cst = stablehlo.constant dense<0.000000e+00> : tensor<f32>
+  %result = "stablehlo.reduce_window"(%arg0, %cst) <{padding = dense<0> : tensor<4x2xi64>, window_dimensions = array<i64: 1, 1, 2, 2>, window_strides = array<i64: 1, 1, 2, 2>}> ({
```
Comment (christopherbate) [inline mlir-tensorrt/test/Dialect/StableHloExt/reify-ranked-shaped-type.mlir:105]: For better coverage, can you please add a test with non-zero padding and positive dilation?

**Review by christopherbate — APPROVED** [review]
One minor request, otherwise LGTM, thanks!

File: mlir-tensorrt/test/Dialect/StableHloExt/reify-ranked-shaped-type.mlir (hunk 1 above)
Comment (yizhuoz004) [reply mlir-tensorrt/test/Dialect/StableHloExt/reify-ranked-shaped-type.mlir]: Included non-zero padding and dilation, thanks
